src/Fluid/Flu_Prepare.cpp

```diff
@@ -53,13 +59,24 @@ void Flu_Prepare( const int lv, const double PrepTime,
    const bool   DE_Consistency      = ( OPT__OPTIMIZE_AGGRESSIVE ) ? DE_Consistency_No : DE_Consistency_Yes;
    const real   MinDens             = ( OPT__OPTIMIZE_AGGRESSIVE ) ? MinDens_No : MIN_DENS;
 
-
 // prepare the fluid array
 #  if ( MODEL == ELBDM )
-   Prepare_PatchData( lv, PrepTime, h_Flu_Array_F_In[0][0], NULL,
-                      FLU_GHOST_SIZE, NPG, PID0_List, _REAL|_IMAG|_PASSIVE, _NONE,
-                      OPT__FLU_INT_SCHEME, INT_NONE, UNIT_PATCHGROUP, NSIDE_26, OPT__INT_PHASE,
-                      OPT__BC_FLU, BC_POT_NONE, MinDens_No, MinPres_No, MinTemp_No, MinEntr_No, DE_Consistency_No );
+#  if ( ELBDM_SCHEME == ELBDM_HYBRID )
+   if ( amr->use_wave_flag[lv] ) {
+#  endif // #  if ( ELBDM_SCHEME == ELBDM_HYBRID )
+      Prepare_PatchData( lv, PrepTime, h_Flu_Array_F_In[0][0], NULL,
+                        FLU_GHOST_SIZE, NPG, PID0_List, _REAL|_IMAG|_PASSIVE, _NONE,
+                        OPT__FLU_INT_SCHEME, INT_NONE, UNIT_PATCHGROUP, NSIDE_26, OPT__INT_PHASE,
+                        OPT__BC_FLU, BC_POT_NONE, MinDens_No, MinPres_No, MinTemp_No, MinEntr_No, DE_Consistency_No );
+
+#  if ( ELBDM_SCHEME == ELBDM_HYBRID )
+   } else  { // if ( amr->use_wave_flag[lv] )
+      Prepare_PatchData( lv, PrepTime, h_Flu_Array_F_In[0][0], NULL,
+                        HYB_GHOST_SIZE, NPG, PID0_List, _DENS|_PHAS|_PASSIVE, _NONE,
+                        OPT__FLU_INT_SCHEME, INT_NONE, UNIT_PATCHGROUP, NSIDE_26, OPT__INT_PHASE,
+                        OPT__BC_FLU, BC_POT_NONE, MinDens_No, MinPres_No, MinTemp_No, MinEntr_No, DE_Consistency_No );
```

Comment:
[Important] I find that the output density can be negative when adopting `INT_SPECTRAL`, suggesting that either the monotonic parameter is set incorrectly in `InterpolateGhostZone()` or the monotonicity implementation is broken in `Int_Spectral()`.

Reply:
- Consider replacing `MinDens_No` with `MinDens` to enable density floor.
- Remove the following lines at the beginning of this function. Let's just trigger a compilation error if `DENS` is not defined for any new `MODEL`.
```
#  if ( MODEL != HYDRO )
   const double MIN_DENS            = -1.0;  // set to an arbitrarily negative value to disable it
#  endif
```
- Raise a warning in `Aux_Check_Parameter()` when enabling `OPT__OPTIMIZE_AGGRESSIVE` (for all models including `HYDRO` and `ELBDM`).